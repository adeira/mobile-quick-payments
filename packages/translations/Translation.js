// @flow

import * as React from 'react';
import { Text } from 'react-native'; // eslint-disable-line no-restricted-imports
import { FormattedMessage } from 'react-intl';
import { type TranslationKeys } from 'mobile-quick-payments-translations';

type Props =
  | {|
      id: TranslationKeys,
    |}
  | {|
      passThrough: string,
    |};

export default (props: Props) => {
  if (props.id !== undefined && props.passThrough !== undefined) {
    throw new Error(
      "You can use only 'id' or 'passThrough' property in translations but not both.",
    );
  }

  if (props.passThrough !== undefined) {
    return <Text>{props.passThrough}</Text>;
  }

  return (
    <FormattedMessage id={props.id}>
      {translatedText => <Text>{translatedText}</Text>}
    </FormattedMessage>
  );
};
