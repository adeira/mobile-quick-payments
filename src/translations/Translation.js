// @flow

import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import { type TranslationKeys } from 'mobile-quick-payments-translations';

type Props = {|
  id: TranslationKeys,
|};

export default ({ id }: Props) => <FormattedMessage id={id} />;
