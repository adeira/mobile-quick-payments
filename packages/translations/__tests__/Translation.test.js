// @flow

import Translation from '../Translation';

it('works with translation ID property', () => {
  // this will call the underlying translation engine
  expect(
    // $FlowExpectedError: string "ID test" is not a valid translation key
    Translation({
      id: 'ID test',
    }),
  ).toMatchSnapshot();
});

it('works with `passThrough` property', () => {
  // this will just return Text tag directly
  expect(
    Translation({
      passThrough: 'passThrough test',
    }),
  ).toMatchSnapshot();
});

it('throws exception for unclear properties', () => {
  expect(() =>
    // $FlowExpectedError: ID and passThrough at the same time is not a valid type
    Translation({
      id: 'ID test',
      passThrough: 'passThrough test',
    }),
  ).toThrowErrorMatchingSnapshot();
});
