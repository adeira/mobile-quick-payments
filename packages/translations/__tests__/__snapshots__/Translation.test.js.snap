// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws exception for unclear properties 1`] = `"You can use only 'id' or 'passThrough' property in translations but not both."`;

exports[`works with \`passThrough\` property 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  passThrough test
</Text>
`;

exports[`works with translation ID property 1`] = `
<FormattedMessage
  id="ID test"
  values={Object {}}
>
  [Function]
</FormattedMessage>
`;
